internal/jobqueue/fsjobqueue/fsjobqueue.go

```diff
@@ -321,7 +325,8 @@ func (q *fsJobQueue) maybeEnqueue(j *job, updateDependants bool) error {
 	if depsFinished {
 		c, exists := q.pending[j.Type]
 		if !exists {
-			return fmt.Errorf("this queue doesn't accept job type '%s'", j.Type)
+			c = make(chan uuid.UUID, channelSize)
+			q.pending[j.Type] = c
```

Comment:
This needs the lock (`q.mu`) held. We do hold it when calling `maybeEnqueue` from `Enqueue` and `FinishJob`, but not in `New`. That's ok, because nothing can run in parallel when the object is just being constructed. However, I think it will make it easier for people reading this code to understand it if we clearly state that the lock must be held during the duration of `maybeEnqueue`.

 (Are Golang's locks recursive?)

Reply:
I added a comment to address this. Thanks!

Reply:
Perfect!